### PR TITLE
elmPackages.{elm-test,elm-analyse}: elm 0.19.1 updates

### PR DESCRIPTION
**Pull request by turboMaCk**
Update elm node based packages after upgrade to 0.19.1 done in https://github.com/NixOS/nixpkgs/pull/71538

Changes:
- elm-test 0.19.1 with elm 0.19.1 compatibility
- elm-analyse 0.16.3 -> 0.16.4 upgrade possible since `node-env.nix` (shared with nodePackages) was updated

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @domenkozar this is hopefully last one from my side for some time.

### CODE REVIEW

File: pkgs/development/compilers/elm/packages/node-packages.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "elm-verify-examples",
     "elm-doc-preview",
     "elm-upgrade",
-    { "elm-analyse": "0.16.3" },
+    "elm-analyse",
```
Comment (turboMaCk) [inline pkgs/development/compilers/elm/packages/node-packages.json]: regenratation of node-env unblocked us

File: pkgs/development/compilers/elm/packages/node-packages.nix [hunk 2]
```diff
@@ -5015,349 +5141,148 @@ in
   elm-test = nodeEnv.buildNodePackage {
     name = "elm-test";
     packageName = "elm-test";
-    version = "0.19.0-rev6";
+    version = "0.19.1";
```
Comment (turboMaCk) [inline pkgs/development/compilers/elm/packages/node-packages.nix]: this is the most important update.

**Comment by jerith666** [comment]
`elm-test` works fine for me with this patch!

**Comment by domenkozar** [comment]
@GrahamcOfBorg build elmPackages

**Comment by turboMaCk** [comment]
be aware that elm-verify-examples still uses the old version of elm-test but this is not specific to nixpkgs - new version was not yet been release to npm either.

**Comment by valberg** [comment]
I'm still getting 0.19.0-rev6 of elm-test - maybe I'm doing things wrong?

It seems as if there is still some 0.19.0-rev6 mentioning in https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/elm/packages/node-packages.nix#L1097-L1105

I'm a total beginner to nix packaging, so might be I'm misunderstanding something :sweat_smile:

**Comment by turboMaCk** [comment]
@valberg more likely you're using channel in which there is still old version. As I said above elm-verify-examples is still depending and using old version of elm test so the place you refer to is its dependecy I belive. This is a new elm-test https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/elm/packages/node-packages.nix#L5141-L5148.

Make sure it's available in channel you're using https://github.com/NixOS/nixpkgs-channels

**Pull request merged**